x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlObjectHandler.java

```diff
@@ -231,35 +231,44 @@ protected void checkIdpSignature(CheckedFunction<Credential, Boolean, Exception>
                 logger.trace("SAML Signature failure caused by", e);
                 return false;
             } catch (Exception e) {
-                logger.warn("Exception while attempting to validate SAML Signature", e);
+                logger.warn("Exception while attempting to validate SAML Signature" + formatIssuer(issuer), e);
                 return false;
             }
         };
         final List<Credential> credentials = idp.getSigningCredentials();
         if (credentials.stream().anyMatch(predicate) == false) {
-            throw samlSignatureException(credentials, signatureText);
+            throw samlSignatureException(issuer, credentials, signatureText);
         }
     }
 
     /**
      * Constructs a SAML specific exception with a consistent message regarding SAML Signature validation failures
      */
-    private ElasticsearchSecurityException samlSignatureException(List<Credential> credentials, String signature, Exception cause) {
+    private ElasticsearchSecurityException samlSignatureException(
+        Issuer issuer,
+        List<Credential> credentials,
+        String signature,
+        Exception cause
+    ) {
         logger.warn(
-            "The XML Signature of this SAML message cannot be validated. Please verify that the saml realm uses the correct SAML"
-                + "metadata file/URL for this Identity Provider"
+            "The XML Signature of this SAML message cannot be validated. Please verify that the saml realm uses the correct SAML "
+                + "metadata file/URL for this Identity Provider{}",
+            formatIssuer(issuer)
         );
         final String msg = "SAML Signature [{}] could not be validated against [{}]";
-        return samlException(msg, cause, signature, describeCredentials(credentials));
+        if (cause != null) {
+            return samlException(msg, cause, signature, describeCredentials(credentials));
+        } else {
+            return samlException(msg, signature, describeCredentials(credentials));
+        }
     }
 
-    private ElasticsearchSecurityException samlSignatureException(List<Credential> credentials, String signature) {
-        logger.warn(
-            "The XML Signature of this SAML message cannot be validated. Please verify that the saml realm uses the correct SAML"
-                + "metadata file/URL for this Identity Provider"
-        );
-        final String msg = "SAML Signature [{}] could not be validated against [{}]";
-        return samlException(msg, signature, describeCredentials(credentials));
+    private ElasticsearchSecurityException samlSignatureException(Issuer issuer, List<Credential> credentials, String signature) {
+        return samlSignatureException(issuer, credentials, signature, null);
+    }
+
+    private String formatIssuer(Issuer issuer) {
+        return issuer != null ? String.format(" [%s]", issuer.getValue()) : "";
```

Comment:
I think we should make it clear that the issuer value is a value we parsed from the request -- esp. since we're logging it in the context where signature verification failed it can potentially be an arbitrary, untrusted value so we should identify that in the log message -- something like `"The issuer included in the SAML message was [{}]."` 

Also, instead of `""` I'd go for an explicit value like `<missing-issuer>` or `<null>`. 

We could call this method `describeIssuer` to match `describeCredentials`

Reply:
One more thought here: 

@jfreden @richard-dennehy: WDYT about limiting the length here with a call to `cleanTruncate` (with a `...`) if we actually truncated to avoid excessively long log lines/exception messages?

Reply:
Could it be more confusing to explicitly describe the issuer as null? In the cases where `issuer` is `null`, all it means is we haven't parsed it from the request yet

Reply:
>Also, instead of "" I'd go for an explicit value like <missing-issuer> or <null>.

For the logout flow, the `issuer` is probably included (so not missing, we just haven't parsed it yet) and if it's not, validation will fail later (if the signature is correct). Because of that, I think `""` is acceptable in this case. My thinking is that `null` as issuer probably won't happen or at least be extremely rare in production and if it happens, validation will fail because it's a required field. 

>@jfreden @richard-dennehy: WDYT about limiting the length here with a call to cleanTruncate (with a ... if we actually truncated) to avoid excessively long log lines/exception messages?

Is that necessary for the issuer? Are you worried the issuer is a very long url?

Reply:
Ah right, good point -- for logout requests we always pass `null`. Okay, in that case let's go with your approach of not including information about the issuer if the value is null (but the more informative message if it available). We could come up with something more nuanced where we call different methods depending on whether it's a null by definition vs. a null by parsing but no need to over-engineer this thing.

Reply:
>We could call this method describeIssuer to match describeCredentials

++

Reply:
> Is that necessary for the issuer? Are you worried the issuer is a very long url?

Worried primarily about processing a value that's arbitrarily malformed since it's not from a trusted source (we log in cases where the signature verification failed). Limiting it to some high-ish value would not impede UX but prevent unforeseen issues around large values. It's a smaller concern since for the cases where the value is available, we've already parsed the entire SAML message and thus done much of the heavy lifting.

Reply:
Thanks for clarifying! Looks like neither opensaml or the saml spec limits the size of the issuer field, so I think adding a `cleanTruncate` call makes sense.